pallets/automation-time/src/lib.rs

```diff
@@ -955,6 +1089,80 @@ pub mod pallet {
 				.saturating_add(<T as Config>::WeightInfo::run_xcmp_task())
 		}
 
+		/// Executes auto compounding delegation and reschedules task on success
+		pub fn run_auto_compound_delegated_stake_task(
+			delegator: T::AccountId,
+			collator: T::AccountId,
+			account_minimum: BalanceOf<T>,
+			frequency: Seconds,
+			task_id: T::Hash,
+			mut task: Task<T>,
+		) -> (Task<T>, Weight) {
+			match Self::compound_delegator_stake(
+				delegator.clone(),
+				collator.clone(),
+				account_minimum,
+				Self::calculate_execution_fee(&task.action, 1),
+			) {
+				Ok(delegation) =>
+					Self::deposit_event(Event::SuccesfullyAutoCompoundedDelegatorStake {
+						task_id,
+						amount: delegation,
+					}),
+				Err(e) => {
+					Self::deposit_event(Event::AutoCompoundDelegatorStakeFailed {
+						task_id,
+						error_message: Into::<&str>::into(e).as_bytes().to_vec(),
+						error: e,
+					});
+					return (
+						task,
+						<T as Config>::WeightInfo::run_auto_compound_delegated_stake_task(),
+					)
+				},
+			}
+
+			let new_execution_times: Vec<UnixTime> =
+				task.execution_times.iter().map(|when| when.saturating_add(frequency)).collect();
+			let _ = Self::reschedule_existing_task(
+				task_id,
+				task.owner_id.clone(),
+				&task.action,
+				new_execution_times.clone(),
+			)
+			.map(|_| {
+				let new_executions_left: u32 = new_execution_times.len().try_into().unwrap();
+				task.executions_left += new_executions_left;
+				new_execution_times.iter().try_for_each(|t| task.execution_times.try_push(*t))
```

Comment:
It didn't seem like the execution times that are attached to a task actually do anything, but it seemed appropriate to keep them in sync anyway.